tests/unit/server/api/routers/v1/test_annotation_configs.py

```diff
@@ -0,0 +1,505 @@
+from copy import deepcopy
+from typing import Any, Optional
+
+import pytest
+from httpx import AsyncClient
+from starlette.status import (
+    HTTP_200_OK,
+    HTTP_400_BAD_REQUEST,
+    HTTP_404_NOT_FOUND,
+    HTTP_409_CONFLICT,
+)
+from strawberry.relay.types import GlobalID
+
+from phoenix.db import models
+from phoenix.db.types.annotation_configs import (
+    AnnotationType,
+    CategoricalAnnotationConfig,
+    CategoricalAnnotationValue,
+    OptimizationDirection,
+)
+from phoenix.server.types import DbSessionFactory
+
+
+@pytest.mark.parametrize(
+    "create_config,update_config",
+    [
+        pytest.param(
+            {
+                "name": "config-name",
+                "type": AnnotationType.CATEGORICAL.value,
+                "description": "Test description",
+                "optimization_direction": OptimizationDirection.MAXIMIZE.value,
+                "values": [
+                    {"label": "Good", "score": 1.0},
+                    {"label": "Bad", "score": 0.0},
+                ],
+            },
+            {
+                "name": "updated-config-name",
+                "type": AnnotationType.CATEGORICAL.value,
+                "description": "Updated description",
+                "optimization_direction": OptimizationDirection.MINIMIZE.value,
+                "values": [
+                    {"label": "Excellent", "score": 1.0},
+                    {"label": "Poor", "score": 0.0},
+                ],
+            },
+            id="categorical",
+        ),
+        pytest.param(
+            {
+                "name": "config-name",
+                "type": AnnotationType.CONTINUOUS.value,
+                "description": "Test description",
+                "optimization_direction": OptimizationDirection.MAXIMIZE.value,
+                "lower_bound": 0.0,
+                "upper_bound": 100.0,
+            },
+            {
+                "name": "updated-config-name",
+                "type": AnnotationType.CONTINUOUS.value,
+                "description": "Updated description",
+                "optimization_direction": OptimizationDirection.MINIMIZE.value,
+                "lower_bound": -10.0,
+                "upper_bound": 10.0,
+            },
+            id="continuous",
+        ),
+        pytest.param(
+            {
+                "name": "config-name",
+                "type": AnnotationType.FREEFORM.value,
+                "description": "Test description",
+            },
+            {
+                "name": "updated-config-name",
+                "type": AnnotationType.FREEFORM.value,
+                "description": "Updated description",
+            },
+            id="freeform",
+        ),
+    ],
+)
+async def test_crud_operations(
```

Comment:
I tend to prefer these "omni tests" to break out into separate tests and not be parametrized, it makes the failure messages easier to parse